pilot/pkg/model/validation.go

```diff
@@ -1547,14 +1547,19 @@ func checkServiceRoleBinding(in *rbac.ServiceRoleBinding) error {
 			errs = appendErrors(errs, fmt.Errorf("do not use * for names or not_names (in rule %d)", i))
 		}
 	}
-	if in.RoleRef != nil && len(in.Actions) > 0 {
-		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef` or `actions` can be specified"))
+	if in.RoleRef != nil && in.Role != "" {
+		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef`, `role`, or `actions` can be specified"))
 		return errs
 	}
-	if in.RoleRef == nil && len(in.Actions) == 0 {
-		errs = appendErrors(errs, fmt.Errorf("`roleRef` or `actions` must be specified"))
+	if (in.RoleRef != nil || in.Role != "") && len(in.Actions) > 0 {
+		errs = appendErrors(errs, fmt.Errorf("only one of `roleRef`, `role`, or `actions` can be specified"))
+		return errs
+	}
+	if in.RoleRef == nil && in.Role == "" && len(in.Actions) == 0 {
+		errs = appendErrors(errs, fmt.Errorf("one of `roleRef`, `role`, or `actions` must be specified"))
```

Comment:
Can you refactor the 3 error message to 1 statement? The same error message is duplicated 3 times.

Reply:
Also please add check for `role == "/"`

Reply:
Will put the error message into a variable. Thanks.

As for the `role == "/"` check, why do we need to check this special case? If the user does this, this is the same as if they provide a service role name that does not exist, since it's empty.

Reply:
In this case, the actual role name is empty and such config should be rejected in validation time.

Reply:
I see. I didn't know that having an empty service role name is invalid (I thought we just ignored it as we wouldn't be able to find a service role with an empty name). Will add a check.

Reply:
Added